go/integration_test/processor/gas_billing_test.go

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package processor
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/Fantom-foundation/Tosca/go/tosca"
+	"github.com/Fantom-foundation/Tosca/go/tosca/vm"
+	op "github.com/ethereum/go-ethereum/core/vm"
+	"go.uber.org/mock/gomock"
+
+	_ "github.com/Fantom-foundation/Tosca/go/processor/floria" // < registers floria processor for testing
+	_ "github.com/Fantom-foundation/Tosca/go/processor/opera"  // < registers opera processor for testing
+)
+
+func TestProcessor_GasBillingEndToEnd(t *testing.T) {
+	tests := map[string]struct {
+		after   WorldState
+		receipt tosca.Receipt
+		result  tosca.Result
+	}{
+		"success": {
+			after: WorldState{
+				{1}: Account{Balance: tosca.NewValue(524000), Nonce: 5},
```

Comment:
I am having a hard time understanding where this final balance is coming from ... so it is difficult to understand the specification covered by this test.

Reply:
I don't think it makes sense to write down the whole gas computation in the unit tests, the idea of this end to end test is to make sure both the reference implementation as well as floria deliver the same results. I added a comment.

Reply:
Something to consider though: unlike in the interpreter case, the reference version of the processor may be buggy.